main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -872,6 +872,70 @@ object ParsedAst {
       */
     case class NativeMethod(sp1: SourcePosition, fqn: Seq[String], args: Seq[ParsedAst.Expression], sp2: SourcePosition) extends ParsedAst.Expression
 
+    /**
+      * NewChannel Expression.
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class NewChannel(sp1: SourcePosition, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * GetChannel Expression.
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param exp the Channel expression.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class GetChannel(sp1: SourcePosition, exp: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * PutChannel Expression
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param exp1 the Channel expression.
+      * @param exp2 the expression to put in the Channel.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class PutChannel(sp1: SourcePosition, exp1: ParsedAst.Expression, exp2: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * CloseChannel Expression.
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param exp the Channel expression.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class CloseChannel(sp1: SourcePosition, exp: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * Spawn Expression.
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param exp the expression.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class Spawn(sp1: SourcePosition, exp: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * Select Expression.
+      *
+      * @param sp1   the position of the first character in the expression.
+      * @param rules the rules of the select expression.
+      * @param sp2   the position of the last character in the expression.
+      */
+    case class Select(sp1: SourcePosition, rules: Seq[SelectRule], sp2: SourcePosition) extends ParsedAst.Expression
```

Comment:
Maybe we should call it SelectChannel, and place it right before/after CloseChannel?